### ISSUE TITLE
Confusion about some code tricks

### ISSUE THREAD

**Issue by h0ngxuanli** [weiyx15/DHGNN#16]
Hi !
  I have been focus on your team's work for about one year.Thank you for your amazing idea and your hypergraph learning team!
  I am a little confused about some details in code,and are they created for better experiment results?:
1. Warm-up parameters:  does the warm-up procedure really work?
![image](https://user-images.githubusercontent.com/68325219/87526716-c4bc1300-c6bd-11ea-9f44-010c473bbbef.png)

  2.HGNN_CONV / GRAPH_CONV before DHGlayer module: I have no idea about why graph convolution operation is conducted before dynamic graph convolution.
![image](https://user-images.githubusercontent.com/68325219/87527134-42801e80-c6be-11ea-8e18-a39df4ddd46b.png)

  3.Sample selection: I am unfamiliar with graph structure data, and if I use DHGNN for image data, should the sample selection be discarded?
![image](https://user-images.githubusercontent.com/68325219/87527163-4c098680-c6be-11ea-80db-64095bf8d086.png)
![image](https://user-images.githubusercontent.com/68325219/87527174-5035a400-c6be-11ea-8687-fa46bd1ee592.png)

  4._structure_select funtcion : Similar to previous question,if I use DHGNN for image data, should the structure selection be discarded? Because I think edge_dict is not a priori knowledge for image data.
![image](https://user-images.githubusercontent.com/68325219/87527412-ad315a00-c6be-11ea-8a02-ccb527c1c0aa.png)
  I am an undergrate without a lot of experience in the hypergraph filed,maybe these questions are a little inaccurate.Thank you a lot for your kindness!
 Best Wishes!!!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by weiyx15** [comment]
Hi! Thank you for your feedback!
1. The performance of first few training steps would be instable without warmup because during first few steps, hypergraph construction are based on current feature maps. If high-quality feature is used, i.e. imgnet pretrained cnn, the warmup steps may be not nesscessary.
2. The raw feature of Cora data is so sparse that a layer is needed for dimension reduction. We use a graph convolution layer as the dimension-reduction layer.
3. Neighborhood sampling aims to keep the input tensor dimension consistent across all vertices. Whether to discard the sampling procedure depends on the form of input data. The sampling is needed when numbers of neighborhood vertices vary across all vertices in a hypergraph. Take the image data as an example. If an image is represented as a vertex, vertices may have different neighborhood sizes. However, if a pixel in an image is represented as a vertex, every vertex has fixed number of neighborhood verteces, in which case neighborhood sampling could be discarded.
4. Same as previous response.

Hope this reply can help you! Best wishes!

**Comment by h0ngxuanli** [comment]
These details really help me a lot!
Thank you for such a detailed response and amazing job!
